### PR TITLE
update GC for release 5.0

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@hicqu Please involve a technical review~

**Comment by hicqu** [comment]
@yiwu-arbug PTAL

File: garbage-collection-configuration.md [hunk 1]
```diff
@@ -75,7 +75,7 @@ update mysql.tidb set VARIABLE_VALUE="24h" where VARIABLE_NAME="tikv_gc_life_tim
 
 - `"distributed"`（默认）：分布式 GC 模式。在此模式下，[Do GC](/garbage-collection-overview.md#do-gc进行-gc-清理) 阶段由 TiDB 上的 GC leader 向 PD 发送 safe point，每个 TiKV 节点各自获取该 safe point 并对所有当前节点上作为 leader 的 Region 进行 GC。此模式于 TiDB 3.0 引入。
 
-- `"central"`：集中 GC 模式。在此模式下，[Do GC](/garbage-collection-overview.md#do-gc进行-gc-清理) 阶段由 GC leader 向所有的 Region 发送 GC 请求。TiDB 2.1 及更早版本采用此 GC 模式。
+- `"central"`：集中 GC 模式。在此模式下，[Do GC](/garbage-collection-overview.md#do-gc进行-gc-清理) 阶段由 GC leader 向所有的 Region 发送 GC 请求。TiDB 2.1 及更早版本采用此 GC 模式，在 TiKV 5.0 及以上版本中已不再支持，设置成这个选项的集群将切换到 distributed 模式。
```
Comment (yiwu-arbug) [inline garbage-collection-configuration.md:78]:
```suggestion
- `"central"`：集中 GC 模式。在此模式下，[Do GC](/garbage-collection-overview.md#do-gc进行-gc-清理) 阶段由 GC leader 向所有的 Region 发送 GC 请求。TiDB 4.x 及更早版本采用此 GC 模式，在 TiKV 5.0 及以上版本中已不再支持，设置成这个选项的集群将切换到 distributed 模式。
```

File: garbage-collection-configuration.md [hunk 2]
```diff
@@ -142,3 +142,43 @@ TiKV 在 3.0.6 版本开始支持 GC 流控，可通过配置 `gc.max-write-byte
 ```bash
 tikv-ctl --host=ip:port modify-tikv-config -m server -n gc.max_write_bytes_per_sec -v 10MB
 ```
+
+## 5.0 中新的 GC 机制
+
+TiKV 在 5.0 版本中引入了新的 GC 机制，在 distribution GC 的基础上，由 RocksDB 的 Compaction 过程来进行 GC，而不再是使用一个单独的 GC worker 线程。这样做的好处是避免了 GC 引起的额外磁盘读取，以及清理掉的旧版本残留大量 tombstone 影响 scan 性能。新机制默认打开，同时支持滚动升级完成之后静默开启。可以由 TiKV 配置文件中的这个开关控制：
```
Comment (yiwu-arbug) [inline garbage-collection-configuration.md:148]:
```suggestion
TiKV 在 5.0 版本中引入了新的 GC 机制，在分布式 GC 模式（distributed GC）的基础上，由 RocksDB 的 Compaction 过程来进行 GC，而不再是使用一个单独的 GC worker 线程。这样做的好处是避免了 GC 引起的额外磁盘读取，以及清理掉的旧版本残留大量删除标记影响顺序扫描性能。新机制默认打开，同时支持滚动升级完成之后静默开启。可以由 TiKV 配置文件中的这个开关控制：
```

File: garbage-collection-configuration.md (hunk 1 above)
Comment (hicqu) [reply garbage-collection-configuration.md:78]: > TiDB 2.1 及更早版本采用此 GC 模式
这个是正确的，3.0 之后默认已经切换到 distributed 模式了，只不过 central 作为配置项仍然保留了下来。

Comment (yiwu-arbug) [reply garbage-collection-configuration.md:78]: ok

Comment (TomShawn) [inline garbage-collection-configuration.md:78]:
```suggestion
- `"central"`：集中 GC 模式。在此模式下，[Do GC](/garbage-collection-overview.md#do-gc进行-gc-清理) 阶段由 GC leader 向所有的 Region 发送 GC 请求。TiDB 2.1 及更早版本采用此 GC 模式。从 TiKV 5.0 起不再支持该模式，设置成这个选项的集群将切换到 `distributed` 模式。
```

File: garbage-collection-configuration.md [hunk 3]
```diff
@@ -142,3 +142,43 @@ TiKV 在 3.0.6 版本开始支持 GC 流控，可通过配置 `gc.max-write-byte
 ```bash
 tikv-ctl --host=ip:port modify-tikv-config -m server -n gc.max_write_bytes_per_sec -v 10MB
 ```
+
+## 5.0 中新的 GC 机制
```
Comment (TomShawn) [inline garbage-collection-configuration.md:146]: 避免出现`新旧`等有时限性的相对词语。
```suggestion
## v5.0 中的 GC 机制
```

File: garbage-collection-configuration.md [hunk 4]
```diff
@@ -142,3 +142,43 @@ TiKV 在 3.0.6 版本开始支持 GC 流控，可通过配置 `gc.max-write-byte
...
+
+TiKV 在 5.0 版本中引入了新的 GC 机制，在分布式 GC 模式（distribution GC）的基础上，由 RocksDB 的 Compaction 过程来进行 GC，而不再是使用一个单独的 GC worker 线程。这样做的好处是避免了 GC 引起的额外磁盘读取，以及清理掉的旧版本残留大量删除标记影响顺序扫描性能。新机制默认打开，同时支持滚动升级完成之后静默开启。可以由 TiKV 配置文件中的这个开关控制：
```
Comment (TomShawn) [inline garbage-collection-configuration.md:148]: + 能否在描述中避免`新`呢？假设一年后有更新的 GC 机制，那么下面的机制就不新了。
+ 可否在 https://github.com/pingcap/docs-cn/blob/master/tikv-configuration-file.md 这里面补充一下该配置项呢？
```suggestion
TiKV 在 5.0 版本中引入了新的 GC 机制。在分布式 GC 模式 (distributed GC) 的基础上，由 RocksDB 的 Compaction 过程来进行 GC，而不再是使用一个单独的 GC worker 线程。这样做的好处是避免了 GC 引起的额外磁盘读取，以及清理掉的旧版本残留大量删除标记影响顺序扫描性能。新 GC 机制默认打开，同时支持滚动升级完成之后静默开启。可以由 TiKV 配置文件中的这个开关控制：
```

**Comment by yiwu-arbug** [comment]
LGTM

File: garbage-collection-configuration.md [hunk 5]
```diff
@@ -142,3 +142,43 @@ TiKV 在 3.0.6 版本开始支持 GC 流控，可通过配置 `gc.max-write-byte
...
+
+```toml
+[gc]
+enable-compaction-filter = true
+```
+
+同时支持在线配置变更。例如：
```
Comment (TomShawn) [inline garbage-collection-configuration.md:155]:
```suggestion
该 GC 支持通过在线配置变更开启：
```

File: garbage-collection-configuration.md [hunk 6]
```diff
@@ -142,3 +142,43 @@ TiKV 在 3.0.6 版本开始支持 GC 流控，可通过配置 `gc.max-write-byte
...
+
+```sql
+show config where type = 'tikv' and name like '%enable-compaction-filter%';
+```
+
+```
```
Comment (TomShawn) [inline garbage-collection-configuration.md:161]:
```suggestion
```sql
```

File: garbage-collection-configuration.md [hunk 7]
```diff
@@ -142,3 +142,43 @@ TiKV 在 3.0.6 版本开始支持 GC 流控，可通过配置 `gc.max-write-byte
...
++------+-------------------+-----------------------------+-------+
+| Type | Instance          | Name                        | Value |
++------+-------------------+-----------------------------+-------+
+| tikv | 172.16.5.37:20163 | gc.enable-compaction-filter | false |
+| tikv | 172.16.5.36:20163 | gc.enable-compaction-filter | false |
+| tikv | 172.16.5.35:20163 | gc.enable-compaction-filter | false |
++------+-------------------+-----------------------------+-------+
+```
+
+```
```
Comment (TomShawn) [inline garbage-collection-configuration.md:171]:
```suggestion
```sql
```

File: garbage-collection-configuration.md [hunk 8]
```diff
@@ -142,3 +142,43 @@ TiKV 在 3.0.6 版本开始支持 GC 流控，可通过配置 `gc.max-write-byte
...
+set config tikv gc.enable-compaction-filter = true;
+show config where type = 'tikv' and name like '%enable-compaction-filter%';
+```
+
+```
```
Comment (TomShawn) [inline garbage-collection-configuration.md:176]:
```suggestion
```sql
```

File: garbage-collection-configuration.md [hunk 9]
```diff
@@ -75,7 +75,7 @@ update mysql.tidb set VARIABLE_VALUE="24h" where VARIABLE_NAME="tikv_gc_life_tim
 
 - `"distributed"`（默认）：分布式 GC 模式。在此模式下，[Do GC](/garbage-collection-overview.md#do-gc进行-gc-清理) 阶段由 TiDB 上的 GC leader 向 PD 发送 safe point，每个 TiKV 节点各自获取该 safe point 并对所有当前节点上作为 leader 的 Region 进行 GC。此模式于 TiDB 3.0 引入。
 
-- `"central"`：集中 GC 模式。在此模式下，[Do GC](/garbage-collection-overview.md#do-gc进行-gc-清理) 阶段由 GC leader 向所有的 Region 发送 GC 请求。TiDB 2.1 及更早版本采用此 GC 模式。
+- `"central"`：集中 GC 模式。在此模式下，[Do GC](/garbage-collection-overview.md#do-gc进行-gc-清理) 阶段由 GC leader 向所有的 Region 发送 GC 请求。TiDB 2.1 及更早版本采用此 GC 模式。从 TiKV 5.0 起不再支持该模式，设置成这个选项的集群将切换到 `distributed` 模式。
```
Comment (TomShawn) [inline garbage-collection-configuration.md:78]:
```suggestion
- `"central"`：集中 GC 模式。在此模式下，[Do GC](/garbage-collection-overview.md#do-gc进行-gc-清理) 阶段由 GC leader 向所有的 Region 发送 GC 请求。TiDB 2.1 及更早版本采用此 GC 模式。从 TiKV 5.0 起不再支持该模式，设置该模式的集群将切换到 `distributed` 模式。
```

File: garbage-collection-configuration.md [hunk 10]
```diff
@@ -142,3 +142,43 @@ TiKV 在 3.0.6 版本开始支持 GC 流控，可通过配置 `gc.max-write-byte
 ```bash
 tikv-ctl --host=ip:port modify-tikv-config -m server -n gc.max_write_bytes_per_sec -v 10MB
 ```
+
+## v5.0 中的 GC 机制
+
+TiKV 在 5.0 版本中引入了新的 GC 机制。在分布式 GC 模式（distributed GC）的基础上，由 RocksDB 的 Compaction 过程来进行 GC，而不再是使用一个单独的 GC worker 线程。这样做的好处是避免了 GC 引起的额外磁盘读取，以及清理掉的旧版本残留大量删除标记影响顺序扫描性能。新 GC 机制默认打开，同时支持滚动升级完成之后静默开启。可以由 TiKV 配置文件中的这个开关控制：
```
Comment (TomShawn) [inline garbage-collection-configuration.md:148]:
```suggestion
TiKV 在 5.0 版本中引入了新的 GC 机制。在分布式 GC 模式 (distributed GC) 的基础上，由 RocksDB 的 Compaction 过程来进行 GC，而不再使用一个单独的 GC worker 线程。这样做的好处是避免了 GC 引起的额外磁盘读取，以及避免清理掉的旧版本残留大量删除标记影响顺序扫描性能。新的 GC 机制默认开启，同时支持滚动升级完成之后静默开启。可以由 TiKV 配置文件中的以下开关控制：
```

File: garbage-collection-configuration.md [hunk 11]
```diff
@@ -142,3 +142,43 @@ TiKV 在 3.0.6 版本开始支持 GC 流控，可通过配置 `gc.max-write-byte
...
+
+```toml
+[gc]
+enable-compaction-filter = true
+```
+
+该 GC 支持通过在线配置变更开启：
```
Comment (TomShawn) [inline garbage-collection-configuration.md:155]:
```suggestion
该 GC 机制可通过在线配置变更开启：
```

File: garbage-collection-configuration.md [hunk 12]
```diff
@@ -142,3 +142,43 @@ TiKV 在 3.0.6 版本开始支持 GC 流控，可通过配置 `gc.max-write-byte
 ```bash
 tikv-ctl --host=ip:port modify-tikv-config -m server -n gc.max_write_bytes_per_sec -v 10MB
 ```
+
+## v5.0 中的 GC 机制
+
+TiKV 在 5.0 版本中引入了新的 GC 机制。在分布式 GC 模式（distributed GC）的基础上，由 RocksDB 的 Compaction 过程来进行 GC，而不再是使用一个单独的 GC worker 线程。这样做的好处是避免了 GC 引起的额外磁盘读取，以及清理掉的旧版本残留大量删除标记影响顺序扫描性能。新 GC 机制默认打开，同时支持滚动升级完成之后静默开启。可以由 TiKV 配置文件中的这个开关控制：
+
```
Comment (TomShawn) [inline garbage-collection-configuration.md:149]:
```suggestion

{{< copyable "" >}}

```

File: garbage-collection-configuration.md [hunk 13]
```diff
@@ -142,3 +142,43 @@ TiKV 在 3.0.6 版本开始支持 GC 流控，可通过配置 `gc.max-write-byte
...
+```toml
+[gc]
+enable-compaction-filter = true
+```
+
+该 GC 支持通过在线配置变更开启：
+
```
Comment (TomShawn) [inline garbage-collection-configuration.md:156]:
```suggestion

{{< copyable "sql" >}}

```

File: garbage-collection-configuration.md [hunk 14]
```diff
@@ -142,3 +142,43 @@ TiKV 在 3.0.6 版本开始支持 GC 流控，可通过配置 `gc.max-write-byte
...
+```sql
++------+-------------------+-----------------------------+-------+
+| Type | Instance          | Name                        | Value |
++------+-------------------+-----------------------------+-------+
+| tikv | 172.16.5.37:20163 | gc.enable-compaction-filter | false |
+| tikv | 172.16.5.36:20163 | gc.enable-compaction-filter | false |
+| tikv | 172.16.5.35:20163 | gc.enable-compaction-filter | false |
++------+-------------------+-----------------------------+-------+
+```
+
```
Comment (TomShawn) [inline garbage-collection-configuration.md:170]:
```suggestion

{{< copyable "sql" >}}

```

File: garbage-collection-configuration.md [hunk 15]
```diff
@@ -142,3 +142,49 @@ TiKV 在 3.0.6 版本开始支持 GC 流控，可通过配置 `gc.max-write-byte
 ```bash
 tikv-ctl --host=ip:port modify-tikv-config -m server -n gc.max_write_bytes_per_sec -v 10MB
 ```
+
+## v5.0 中的 GC 机制
```
Comment (TomShawn) [inline garbage-collection-configuration.md:146]:
```suggestion
## GC in Compaction Filter 机制
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**